### PR TITLE
add wavelet computation

### PR DESCRIPTION
**Pull request by LaureBrenot**
starts with:
```
msnoise new_jobs --hpc STACK:WCT
msnoise cc dvv compute_wct
```
Careful:  ```msnoise db init``` has to be done after the modifications to consider the new parameters in the configurations.
Plus, ```pip install pywct```

If accepted functions of s08compute_wct.py can be moved to api.py

Based on A. Yates and R. De Plaen works

### CODE REVIEW

File: s08compute_wct.py [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+        tindex = np.where(((tvec >= -lag_max) & (tvec <= -lag_min)) | ((tvec >= lag_min) & (tvec <= lag_max)))[0] # Index of the coda
+
+        if len(tvec)>2: # check time vector size
+            if not np.any(wcoh[ifreq]): # check non-empty dt array
+                continue
+            c = np.nanmean(wcoh[ifreq][tindex])
+            coh[ii] = c
+
+        else:
+            print('not enough points to compute average coherence') #not sure why it would ever get here, but just in case.
```
Comment (ThomasLecocq) [inline s08compute_wct.py:70]: replace print() by logging.debug() or .warning()

File: s08compute_wct.py [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+    # Convolve the coefficients with a moving average smoothing filter across scales.
+    H = 1 / ns * np.ones((ns, 1))
+
+    cfs = conv2(cfs, H)
+    return cfs
+
+
+## nextpow2 function
+# Returns the exponents p for the smallest powers of two that satisfy the relation  : 2**p >= abs(x)
+def nextpow2(x):
```
Comment (ThomasLecocq) [inline s08compute_wct.py:104]: would be good/faster to use https://docs.scipy.org/doc/scipy/reference/generated/scipy.fft.next_fast_len.html#scipy.fft.next_fast_len

File: s08compute_wct.py [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+        Smallest frequency
+        Default value is 0.1 Hz
+    freqmax : float,
+        Highest frequency
+        Default value is 8.0 Hz
+    nptsfreq : int,
+        Number of frequency points between freqmin and freqmax.
+        Default value is 100 points
+       
+    Returns
```
Comment (ThomasLecocq) [inline s08compute_wct.py:189]: lacks the return docstring

File: s08compute_wct.py [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+    """
+    # Choosing a Morlet wavelet with a central frequency w0 = 6
+    mother = wavelet.Morlet(6.)
```
Comment (ThomasLecocq) [inline s08compute_wct.py:192]: would that be a parameter to be configured by users ? (have zero idea what it does). 
Is Morlet the only choice too ?

File: s08compute_wct.py [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+    ds = xr.Dataset({
+        'dvv': dvv_da,
+        'err': err_da,
+        'coh': coh_da
+    })
+
+    # Save the dataset to a NetCDF file
+    ds.to_netcdf(fn)
+    
+    print(f"Saved WCT data to {fn}")
```
Comment (ThomasLecocq) [inline s08compute_wct.py:289]: logging.debug()

File: s08compute_wct.py [hunk 6]
```diff
@@ -0,0 +1,425 @@
...
+            # Percentage of non-zero weights
+            nzc_perc = np.count_nonzero(w[tindex]) / len(tindex)
+
+            if nzc_perc >= min_nonzero:
+                m, em = linear_regression(tvec[tindex], delta_t[ifreq][tindex], w[tindex], intercept_origin=True)
+                dvv[ii], err[ii] = -m, em
+            else:
+                dvv[ii], err[ii] = np.nan, np.nan
+        else:
+            print('Not enough points to estimate dv/v for WCT')
```
Comment (ThomasLecocq) [inline s08compute_wct.py:151]: logging.warning()

File: s08compute_wct.py [hunk 7]
```diff
@@ -0,0 +1,425 @@
+"""
+WCT
```
Comment (ThomasLecocq) [inline s08compute_wct.py:2]: would ideally need some more documentation, but I reckon it could be in another PR (I'm working on a automated documentation builder that would allow checking the PRs)

File: s08compute_wct.py [hunk 8]
```diff
@@ -0,0 +1,425 @@
...
+    res = np.ceil(np.log2(x))
+    return res.astype('int')
+
+## conv2 function
+# Returns the two-dimensional convolution of matrices x and y
+def conv2(x, y, mode='same'):
+    return np.rot90(convolve2d(np.rot90(x, 2), np.rot90(y, 2), mode=mode), 2)
+
+
+def get_dvv(freqs, tvec, WXamp, Wcoh, delta_t, lag_min=5, coda_cycles=20, mincoh=0.5, maxdt=0.2,
```
Comment (ThomasLecocq) [inline s08compute_wct.py:114]: that name exists in api.py and is too generic, needs to be wct_specific?

**Review by ThomasLecocq — CHANGES_REQUESTED** [review]
Looks super !! thanks a lot!

File: s08compute_wct.py (hunk 4 above)
Comment (LaureBrenot) [reply s08compute_wct.py:192]: Yes, it could be. There are four different mother wavelet classes (Morlet, Paul, DOG, MexicanHat), and they have their own default values. I could add parameters wavelet_type and wavelet_scale in the config
